pkg/server/server.go

```diff
@@ -347,3 +351,87 @@ func (host *yamlLanguageHost) GenerateProgram(
 		Diagnostics: rpcDiagnostics,
 	}, nil
 }
+
+func (host *yamlLanguageHost) GeneratePackage(ctx context.Context, req *pulumirpc.GeneratePackageRequest) (*pulumirpc.GeneratePackageResponse, error) {
+	// YAML doesn't generally have "SDKs" per-se but we can write out a "lock file" for a given package name and
+	// version, and if using a parameterized package this is necessary so that we have somewhere to save the parameter
+	// value.
+
+	if len(req.ExtraFiles) > 0 {
+		return nil, errors.New("overlays are not supported for YAML")
+	}
+
+	loader, err := schema.NewLoaderClient(req.LoaderTarget)
+	if err != nil {
+		return nil, err
+	}
+
+	var spec schema.PackageSpec
+	err = json.Unmarshal([]byte(req.Schema), &spec)
+	if err != nil {
+		return nil, err
+	}
+
+	pkg, diags, err := schema.BindSpec(spec, loader)
+	if err != nil {
+		return nil, err
+	}
+	rpcDiagnostics := plugin.HclDiagnosticsToRPCDiagnostics(diags)
+	if diags.HasErrors() {
+		return &pulumirpc.GeneratePackageResponse{
+			Diagnostics: rpcDiagnostics,
+		}, nil
+	}
+
+	// Generate the a package lock file in the given directory. This is just a simple YAML file that contains the name,
+	// version, and any parameter values.
+	lock := packages.PackageDecl{
+		PackageDeclarationVersion: 1,
+	}
+	// The format of the lock file differs based on if this is a parameterized package or not.
+	if pkg.Parameterization == nil {
+		lock.Name = pkg.Name
+		if pkg.Version != nil {
+			lock.Version = pkg.Version.String()
+		}
+		lock.DownloadURL = pkg.PluginDownloadURL
+	} else {
+		lock.Name = pkg.Parameterization.BaseProvider.Name
+		lock.Version = pkg.Parameterization.BaseProvider.Version.String()
```

Comment:
I know it would be invalid, but are we certain at this point that `Version` is not `nil`? Guessing `BindSpec` guarantees it so no need to be defensive? Or perhaps `Version` is not a pointer here, in which case ignore me :see_no_evil:

Reply:
It's not a pointer here

Reply:
Oh thank deity -- I was worried I wouldn't get to look like an idiot on this review.